spi/data-plane/data-plane-spi/src/test/java/org/eclipse/edc/connector/dataplane/spi/pipeline/StreamFailureTest.java

```diff
@@ -0,0 +1,43 @@
+/*
+ *  Copyright (c) 2024 Bayerische Motoren Werke Aktiengesellschaft (BMW AG)
+ *
+ *  This program and the accompanying materials are made available under the
+ *  terms of the Apache License, Version 2.0 which is available at
+ *  https://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  SPDX-License-Identifier: Apache-2.0
+ *
+ *  Contributors:
+ *       Bayerische Motoren Werke Aktiengesellschaft (BMW AG) - Initial implementation
+ *
+ */
+
+package org.eclipse.edc.connector.dataplane.spi.pipeline;
+
+import org.junit.jupiter.api.Test;
+
+import java.util.Collections;
+import java.util.List;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.eclipse.edc.connector.dataplane.spi.pipeline.StreamFailure.Reason.NOT_FOUND;
+
+public class StreamFailureTest {
+    private static final String failureMessage = "failure message";
+
+    @Test
+    void verify_with_failureDetail_message_starts_with_reason() {
+        assertThat(new StreamFailure(List.of(failureMessage), NOT_FOUND)
+                .getFailureDetail()
+                .startsWith(NOT_FOUND.toString())
+        ).isTrue();
```

Comment:
this test can be written in a crispier way leveraging on the assertJ dsl:

```
var failure = new StreamFailure(List.of(failureMessage), NOT_FOUND);

assertThat(failure.getFailureDetail()).startsWith(NOT_FOUND.toString());
```

same for the other